pkg/kubectl/cmd/version.go

```diff
@@ -46,36 +54,104 @@ func NewCmdVersion(f cmdutil.Factory, out io.Writer) *cobra.Command {
 	}
 	cmd.Flags().BoolP("client", "c", false, "Client version only (no server required).")
 	cmd.Flags().BoolP("short", "", false, "Print just the version number.")
+	cmd.Flags().String("output", "", "output format, options available are yaml and json")
 	cmd.Flags().MarkShorthandDeprecated("client", "please use --client instead.")
 	return cmd
 }
 
 func RunVersion(f cmdutil.Factory, out io.Writer, cmd *cobra.Command) error {
-	v := fmt.Sprintf("%#v", version.Get())
-	if cmdutil.GetFlagBool(cmd, "short") {
-		v = version.Get().GitVersion
-	}
+	vo := VersionObj{nil, nil}
+
+	cvg := version.Get()
+	vo.ClientVersion = &cvg
+
+	switch of := cmdutil.GetFlagString(cmd, "output"); of {
+	case "":
+		if cmdutil.GetFlagBool(cmd, "short") {
+			fmt.Fprintf(out, "Client Version: %s\n", vo.ClientVersion.GitVersion)
+			if cmdutil.GetFlagBool(cmd, "client") {
+				return nil
+			}
+
+			serverVersion, err := retrieveServerVersion(f)
```

Comment:
retrieve the server version once at the top of the function, just save the serverErr and don't return early. the function should look like this:

1. get client version
2. if needed, get server version
3. switch on output type and output
4. return serverErr if we got an error